pkg/ipam/crd_eni.go

```diff
@@ -0,0 +1,199 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ipam
```

Comment:
Perhaps it would make sense to have this file in its own package so that other operators from cloud providers do not import `"github.com/cilium/cilium/pkg/aws/eni/types"`?

Reply:
The commit does not introduce any new dependency on cloud providers, `pkg/ipam` already imports `"github.com/cilium/cilium/pkg/aws/eni/types"`:

https://github.com/cilium/cilium/blob/59ccd7a4067d0f707e2c2fda51b239043af9835e/pkg/ipam/crd.go#L28

Reply:
Also note: This is agent code, not operator code.